java/code/src/com/redhat/rhn/frontend/xmlrpc/user/UserHandler.java

```diff
@@ -1141,4 +1144,63 @@ public int setErrataNotifications(User loggedInUser, String login, Boolean value
         targetUser.setEmailNotify(BooleanUtils.toIntegerObject(value));
         return 1;
     }
+
+    /**
+     * @param user The current user
+     * @param unread Unread notifications
+     * @return Returns a list of notifications
+     * @apidoc.doc Get all notifications from a user.
+     * @apidoc.param #session_key()
+     * @apidoc.param #param_desc("boolean", "unread", "Read notifications.")
+     * @apidoc.returntype #return_notifications()
+     */
+    @ReadOnly
+    public List<UserNotification> getNotifications(User user, boolean unread) {
+        List<UserNotification> notifications;
+
+        if (unread) {
+            notifications = UserNotificationFactory.listUnreadByUser(user);
+        } else {
+            notifications = UserNotificationFactory.listAllByUser(user);
+        }
+        return notifications;
+    }
+
+    /**
+     * @param userNotification The notification
+     * @return Returns 1 if successful
+     * @apidoc.doc Makes a notification raed
+     * @apidoc.param #param_desc("userNotification", "notification", "The target notification.")
+     * @apidoc.returntype #return_int_success()
+     */
+    public int makeNotificationRead(User user, UserNotification notification) {
```

Comment:
I think having just the ID as input would be easier to use (considering both XMLRPC and HTTP)

Reply:
Also, accepting a collection of notifications instead of a single one would cover more cases.